### PR TITLE
Rename ParentRef to be consistent with other reference types.

### PR DESCRIPTION
**Pull request by jpeach**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

Across the API, the object reference types are all called
`FooBarReference`. To be consistent with this pattern, `ParentRef`
should be called `ParentReference`.

A few issues with this PR

1. Not convinced that this is a compatible change for code that consumes this module
2. deepcopy-gen generates code for both the type and its alias, which is deeply wrong

**Which issue(s) this PR fixes**:

N/A

**Does this PR introduce a user-facing change?**:

```release-note
ParentRef type renamed to ParentReference.
```

### CODE REVIEW

File: apis/v1alpha2/shared_types.go [hunk 1]
```diff
@@ -86,6 +86,9 @@ type ParentRef struct {
 	SectionName *SectionName `json:"sectionName,omitempty"`
 }
 
+// ParentRef is a backwards compatibility alias for ParentReference.
+type ParentRef = ParentReference
```
Comment (hbagdi) [inline apis/v1alpha2/shared_types.go:90]: I think this is upsetting the code generation?

Comment (jpeach) [reply apis/v1alpha2/shared_types.go:90]: Yup

> deepcopy-gen generates code for both the type and its alias, which is deeply wrong

Comment (hbagdi) [reply apis/v1alpha2/shared_types.go:90]: I think not having this type of backward compatibility with Go APIs like this is reasonable. Larger changes may pose problems but this is okay.
Sensitive area so would like to hear others.

Comment (howardjohn) [reply apis/v1alpha2/shared_types.go:90]: +1, I am not too concerned with go compatibility. Especially when its a trivial change

Comment (robscott) [reply apis/v1alpha2/shared_types.go:90]: Agreed, this seems fine. Upstream kubernetes has also not been concerned about go compat historically.

Comment (jpeach) [reply apis/v1alpha2/shared_types.go]: Sounds great! I removed the compatibility alias, so let's see what CI says about that.

**Comment by hbagdi** [comment]
LGTM.
Grepped for places that could need updates but didn't find any.
Leaving the final approval for another maintainer.

**Comment by robscott** [comment]
Thanks!

/lgtm

**Pull request merged**